connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -551,6 +551,13 @@ class SparkConnectPlanner(session: SparkSession) {
             .Product(transformExpression(fun.getArgumentsList.asScala.head))
             .toAggregateExpression())
 
+      case "in" =>
```

Comment:
if we don't intercept `in` here, it will fail with
```
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "[WRONG_NUM_ARGS.WITHOUT_SUGGESTION] Invalid number of arguments for the function `in`. Please, refer to 'https://spark.apache.org/docs/latest/sql-ref-functions.html' for a fix."
        debug_error_string = "UNKNOWN:Error received from peer ipv6:%5B::1%5D:15002 {grpc_message:"[WRONG_NUM_ARGS.WITHOUT_SUGGESTION] Invalid number of arguments for the function `in`. Please, refer to \'https://spark.apache.org/docs/latest/sql-ref-functions.html\' for a fix.", grpc_status:2, created_time:"2022-12-15T17:17:23.380565+08:00"}"
>
```

This is the second case (the first one is `when`) that an expression has already been registered in `FunctionRegistry`, but do not have a constructor matching `UnresolvedFunction`.

I match them here instead of adding an extra constructor.